cmd/habitat/community/handlers.go

```diff
@@ -224,6 +237,13 @@ func (m *Manager) CommunityJoinHandler(w http.ResponseWriter, r *http.Request) {
 		log.Error().Msgf("error writing final community join response to websocket: %s", err)
 		return
 	}
+
+	peer, _, err := compass.DecomposeNodeMultiaddr(commReq.AcceptingNodeAddr)
+	if err != nil {
+		log.Error().Msgf("error decomposing accepting node multiaddr: %s", err)
+		return
+	}
+	m.node.DataProxy.AddPeerNode(string(peer.String()), commReq.AcceptingNodeAddr)
```

Comment:
We should add peer nodes in the state change executor. There should be a defined function for when a node is added already in `executor.go`. The reason to do it there is that this handler is only run on the leader of the cluster, while that executor will be run by every node.